vibe_code_apps_19/tests/test_bootstrap.py

```
@@ -57,3 +58,11 @@ def test_resolve_env_bootstrap(tmp_path: Path, monkeypatch):
     loaded = read_bootstrap()
     assert loaded is not None
     assert loaded["auto_run_rules"] is False
+
+
+def test_agent_bootstrap_allowed_cloud_vs_local(monkeypatch):
+    monkeypatch.delenv("VIBE19_BOOTSTRAP", raising=False)
+    assert agent_bootstrap_allowed(is_cloud=False) is True
+    assert agent_bootstrap_allowed(is_cloud=True) is False
+    monkeypatch.setenv("VIBE19_BOOTSTRAP", "/tmp/explicit.json")
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp/explicit.json"

(S108)